static/app/utils/useHoverOverlay.tsx

```diff
@@ -266,14 +266,35 @@ function useHoverOverlay(
    */
   const wrapTrigger = useCallback(
     (triggerChildren: React.ReactNode) => {
-      const props = {
+      const makeProps = (
+        componentProps: Partial<
+          Pick<
+            React.HTMLAttributes<any>,
+            'onFocus' | 'onBlur' | 'onPointerEnter' | 'onPointerLeave'
+          >
+        >
+      ) => ({
+        // !!These props are always overriden!!
         'aria-describedby': describeById,
         ref: setTriggerElement,
-        onFocus: handleMouseEnter,
-        onBlur: handleMouseLeave,
-        onPointerEnter: handleMouseEnter,
-        onPointerLeave: handleMouseLeave,
-      };
+        // The following props are composed from the componentProps trigger props
+        onFocus: (e: React.FocusEvent<any>) => {
+          handleMouseEnter();
+          componentProps.onFocus?.(e);
+        },
+        onBlur: (e: React.FocusEvent<any>) => {
+          handleMouseLeave();
+          componentProps.onBlur?.(e);
+        },
+        onPointerEnter: (e: React.PointerEvent<any>) => {
+          handleMouseEnter();
+          componentProps.onPointerEnter?.(e);
+        },
+        onPointerLeave: (e: React.PointerEvent<any>) => {
+          handleMouseLeave();
+          componentProps.onPointerLeave?.(e);
+        },
+      });
```

Comment:
I wonder if we could use this https://react-spectrum.adobe.com/react-aria/mergeProps.html